main.js

```diff
@@ -1,82 +1,157 @@
 // use the variables below to run Math function
-let val1;
-let op;
+let val1 = '';
+let val2 = '';
+let op = '';
+let history = ' ';
+let lastAnsw;
+let lastHist;
+let prevAns;
+
 
 function inputNum(number) {
-    // if value is 0 then it is replaced by selected value. Otherwise adds to display value
-    if (document.getElementById('calc').value == 0 || isNaN(document.getElementById('calc').value)) {
-        if (document.getElementById('calc').value == 0 + '.') {
-            // nested if to account for 0. being treated as NaN
-            document.getElementById('calc').value = document.getElementById('calc').value + number;
+    const currentNumber = document.getElementById('calc').value;
+    // console.log(val1);
+    // If op is undefined then work on creating var1 //
+    if ( lastAnsw !== undefined ) {
+        if ( op !== '' ) {
+            lastAnsw = undefined;
+            document.getElementById('calc').value = number;
+            val2 = document.getElementById('calc').value;
         } else {
```

Comment:
If you don't need to do anything in this else lets go ahead and remove the whole else block!